src/main/java/org/semver4j/internal/range/processor/HyphenProcessor.java

```diff
@@ -24,10 +26,12 @@
  * </ul>
  */
 public class HyphenProcessor implements Processor {
+    @NotNull
```

Comment:
That's not necessary in favor of field object assignment.

Reply:
Reasons 1 & 6